File: Axiom/Internal/Ast/ReferenceExpression.cs
namespace Axiom.Internal.Ast
{
    internal class ReferenceExpression : Expression
    {
        internal enum ReferenceType
        {
            This, Base
        }

        public ReferenceExpression(ReferenceType type)
        {
            Type = type;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public ReferenceType Type { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/ReferenceExpression.java
package com.moybl.axiom.ast;

public class ReferenceExpression implements Expression {

	public enum Kind {
		THIS, BASE
	}

	private Kind kind;

	public ReferenceExpression(Kind kind) {
		this.kind = kind;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Kind getKind() {
		return kind;
	}

}
